after-effects/src/aegp/suites/utility.rs

```diff
@@ -62,10 +62,20 @@ impl UtilitySuite {
 
     /// Returns an [`PluginId`], which effect plug-ins can then use in calls to many functions throughout the AEGP API.
     /// Effects should only call this function once, during [`Command::GlobalSetup`], and save the [`PluginId`] for later use.
-    /// The first parameter can be any value, and the second parameter should be the plug-in's match name.
-    pub fn register_with_aegp(&self, global_refcon: Option<*mut std::ffi::c_void>, plugin_name: &str) -> Result<PluginId, Error> {

```

Comment:
This PR is cleaned up and ready for review/merge, but I want to single this out as I believe it will be a controversial change. 

Instead of taking an Option<*void mut>  I decided to split the `register_with_aegp` into two functions. One which takes an owned T, constrained by a marker trait that documents that the user should only use it once - and the other which takes nothing. All other API's which rely on extracting that type from a raw pointer are constrained by the same unsafe trait. This is as close as I can get to real type safety on this API.